# Load data and Setup paths

In [1]:
from prepare_data.data_loader import load_coco_json, get_dataframes    #from test_1 this normally works
# Import your new functions from data_explorer
from prepare_data.data_explorer import get_basic_stats, show_categories, count_annotations_per_image, count_annotations_per_category

# Setup paths
path = "Data/raw/_annotations.coco.json"

# Load data
data = load_coco_json(path)
df_imgs, df_anns = get_dataframes(data)



In [2]:
print(f"There's {len(df_imgs)} images!")

There's 500 images!


In [ ]:
# CALL the functions from data_explorer to see results 
print("-- EXPLORATION --" \
"")
get_basic_stats(df_imgs, df_anns)
show_categories(data)
count_annotations_per_image(df_anns)
count_annotations_per_category(df_anns, data)

-- EXPLORATION --
Total Images: 500
Total Annotations: 959
Categories found: ['wildfire', 'fire']
--- Annotations per Image ---
Average: 1.95
Max in one image: 10
Min in one image: 1
--- Distribution per Category ---
category_name
fire    959
Name: count, dtype: int64


In [5]:
print("Raw Category IDs in annotations:", df_anns['category_id'].unique())

Raw Category IDs in annotations: [1]


In [7]:
for cat in data['categories']:
    print(f"ID: {cat['id']} | Name: {cat['name']} | Supercategory: {cat.get('supercategory', 'None')}")

ID: 0 | Name: wildfire | Supercategory: none
ID: 1 | Name: fire | Supercategory: wildfire


"fire" belongs to "wildfire,"  essentially saying: "This is a fire, which is a type of wildfire."

Insights: 

Our initial analysis of the 500 satellite images reveals a total of 959 annotations. On average, each image contains 1.95 fire zones, with some images containing up to 10 distinct burned areas. This confirms that the model must be capable of multi-object detection within a single frame.

The Labeling Paradox:
While the dataset metadata defines two categories—"wildfire" (ID 0) and "fire" (ID 1)—our distribution analysis shows that 100% of the annotations (959/959) are tagged as "fire." The "wildfire" category exists only as a parent supercategory in the JSON structure and contains zero active labels.

Strategic Conclusion:
Given that we have a one-week development window, we have decided to simplify the project into a single-class detection task. Since "fire" is a sub-type of "wildfire," focusing exclusively on the "fire" label allows us to maximize model precision and avoid the technical debt of handling an empty class. This approach ensures a functional, high-performance model within the project's time constraints.